workflows/cloudify_system_workflows/snapshots/snapshot_restore.py

```diff
@@ -308,3 +274,104 @@ def _get_tasks_graph(dep_ctx, blueprint, deployment):
                 'groups': deployment['groups']
             }
         )
+
+
+class SnapshotRestoreValidator(object):
```

Comment:
The errors here should be pretty self explanatory. Do make sure I haven't forgotten anything, though